modules/ppcp-googlepay/resources/js/GooglepayButton.js

```diff
@@ -525,6 +525,11 @@ class GooglepayButton extends PaymentButton {
 						updatedData.total,
 						updatedData.shipping_fee
 					);
+
+					// This page contains a real cart and potentially a form for shipping options.
+					this.syncShippingOptionWithForm(
```

Comment:
When the page contains a real cart, check if it contains form elements with shipping-method details.

This method updates the WooCommerce form:
1. Select the same shipping option as is used in the Google Pay form.
2. Trigger an Ajax call to update the total amount in the WooCommerce form.